File: frontend/download_maps.js
const fs = require('fs');
const https = require('https');
const path = require('path');

const publicDataDir = path.join(__dirname, 'public', 'data');
if (!fs.existsSync(publicDataDir)) {
    fs.mkdirSync(publicDataDir, { recursive: true });
}

function download(url, dest) {
    return new Promise((resolve, reject) => {
        console.log(`Downloading ${url} -> ${dest}`);
        const req = https.get(url, {
            headers: { 'User-Agent': 'Mozilla/5.0' }
        }, (res) => {
            if (res.statusCode === 301 || res.statusCode === 302) {
                return download(res.headers.location, dest).then(resolve).catch(reject);
            }
            if (res.statusCode !== 200) {
                reject(new Error(`Failed with status ${res.statusCode}`));
                return;
            }
            const file = fs.createWriteStream(dest);
            res.pipe(file);
            file.on('finish', () => {
                file.close(resolve);
            });
        });
        req.on('error', reject);
    });
}

async function run() {
    await download('https://unpkg.com/world-atlas@2.0.2/countries-110m.json', path.join(publicDataDir, 'world-110m.json'));
    await download('https://raw.githubusercontent.com/yeyusiz/kazakhstan-geojson/master/kazakhstan.geojson', path.join(publicDataDir, 'kz-regions.json'));
    console.log('Maps downloaded successfully!');
}

run().catch(console.error);

File: frontend/download_maps.py
import urllib.request
import os
import ssl
import json

os.makedirs('public/data', exist_ok=True)

# Ignore SSL errors
ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE

def download_file(url, out_path):
    # Use Mozilla User-Agent to bypass Highcharts 429
    req = urllib.request.Request(
        url, 
        data=None, 
        headers={
            'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/35.0.1916.47 Safari/537.36'
        }
    )
    print(f"Downloading {url} to {out_path}...")
    with urllib.request.urlopen(req, context=ctx) as response:
        data = response.read()
        with open(out_path, 'wb') as f:
            f.write(data)
    print("Done.")

download_file('https://unpkg.com/world-atlas@2.0.2/countries-110m.json', 'public/data/world-110m.json')
download_file('https://code.highcharts.com/mapdata/countries/kz/kz-all.topo.json', 'public/data/kz-regions.json')
